### PR TITLE
ts-migration/no-large-snapshots

### PR DESCRIPTION
**Pull request by G-Rath**
Based off #375

### CODE REVIEW

File: src/rules/no-large-snapshots.ts [hunk 1]
```diff
@@ -78,28 +103,34 @@ export default {
...
       };
     } else if (context.getFilename().endsWith('.js')) {
       return {
         CallExpression(node) {
-          const propertyName =
-            node.callee.property && node.callee.property.name;
           if (
-            propertyName === 'toMatchInlineSnapshot' ||
-            propertyName === 'toThrowErrorMatchingInlineSnapshot'
+            'property' in node.callee &&
```
Comment (G-Rath) [inline src/rules/no-large-snapshots.ts]: @SimenB I know what you're going to say - I'm going to implement varargs in it's own PR, as it's a waterfall change that while relatively straightforward, does require a bit of its own work due to subtyping.

The main reason I replaced the array here is this way `isSupportedAccessor` has full coverage; otherwise I'd have left it as `includes` & done the varargs thing in another PR.

File: src/rules/tsUtils.ts [hunk 2]
```diff
@@ -108,6 +108,128 @@ export const isStringNode = <V extends string>(
...
+ * @param {AccessorNode<S>} accessor
+ *
+ * @return {S}
+ *
+ * @template S
+ */
+export const getAccessorValue = <S extends string = string>(
+  accessor: AccessorNode<S>,
+): S =>
+  /* istanbul ignore next */
```
Comment (G-Rath) [inline src/rules/tsUtils.ts]: This is b/c we have no rules that use this on an `Identifier` so coverage complains, *but* `no-large-snapshots` uses `isSupportedAccessor` on an `Identifier`, meaning `AccessorNode` needs to be a union, and so TypeScript requires us to handle both branches.

It's a catch-22 :)

**Comment by G-Rath** [comment]
FYI I'm in the middle of rebasing this one :)

File: src/rules/tsUtils.ts (hunk 2 above)
Comment (SimenB) [reply src/rules/tsUtils.ts]: you solved this another way? Seems to be gone

**Comment by SimenB** [comment]
>FYI I'm in the middle of rebasing this one :)

Heh, new conflict 😅

**Comment by G-Rath** [comment]
>Heh, new conflict 😅

 hehe yeah I'm going to finish the second round of rebasing, and then go to bed, cause I've got work in 5 hours.

I'll look to respond to your other review stuff "tomorrow", but in general from what I've read yes to everything :P

File: src/rules/tsUtils.ts (hunk 2 above)
Comment (G-Rath) [reply src/rules/tsUtils.ts]: I solved it by using it - this was never needed in this branch, but slipped in via the rebase

**Comment by G-Rath** [comment]
Provided the tests push, I'm going to push up the rebased version.

@SimenB if you've got the time, feel free to rebase the commits to make future rebasing easier while I'm sleeping.

Right now it's actually pretty horrible, I think in part b/c there's a number of now redundant commits that are very steppy i.e rename method, move method, remove method -> end result is method that now already exists in master.

Otherwise, I can try and remove them via rebase when I get to work.

Cheers

**Comment by SimenB** [comment]
I can rebase #363 after landing this and see if I can clear out a few of the "chore" commits

File: src/rules/tsUtils.ts (hunk 2 above)
Comment (G-Rath) [reply src/rules/tsUtils.ts]: Sorry thought this was #363 - my tests locally still require this for this PR?

**Comment by G-Rath** [comment]
@SimenB should be good to go - I've id'd a couple of changes that can be rebased away in #363 so hopefully that helps.

File: src/rules/tsUtils.ts (hunk 2 above)
Comment (SimenB) [reply src/rules/tsUtils.ts]: this line doesn't exist anymore
